### PR TITLE
fix spellcheck glob and spellcheck issues that follow

### PR DESCRIPTION
**Pull request by tuminoid**
Since we're running the check in sh, it does not have globstar option to handle '**/*.md' properly. It only checked exactly one level deep markdown files. Spellcheck handles globs itself, so just adding quotes is enough to fix it.

Second commit fixes all the spellcheck issues found by this fix.

Fixes: #479

### CODE REVIEW

**Comment by tuminoid** [comment]
Failing spellchecks to be fixed in follow-up commit in this PR can be seen: https://prow.apps.test.metal3.io/view/s3/prow-logs/pr-logs/pull/metal3-io_metal3-docs/480/spellcheck/1848236935080841216

**Comment by tuminoid** [comment]
This needs design approval as well as it fixes 59 files for typos.

/cc @dtantsur @kashifest @lentzi90

**Comment by tuminoid** [comment]
Rebased as we merged another doc with typos to fix.

**Review by lentzi90 — COMMENTED** [review]
/approve

File: design/baremetal-operator/bmc-events.md [hunk 1]
```diff
@@ -79,15 +81,15 @@ status:
 
 - A BMCEventSubscription resource represents a subscription to the events generated
   by a specific BMC.
-- Ironic will manage configuring the subscription using a vendor passthru API.
```
Comment (dtantsur) [inline design/baremetal-operator/bmc-events.md:82]: On the fence here: Ironic actually calls it "passthru".

File: design/baremetal-operator/bmc-events.md [hunk 2]
```diff
@@ -136,7 +138,9 @@ implementations (if they exist at all).
 
 ## References
 
-- [Ironic Vendor Passthru for Subscriptions](https://storyboard.openstack.org/#!/story/2009061)
+<!-- cSpell:ignore Supermicro,DMTF -->
```
Comment (dtantsur) [inline design/baremetal-operator/bmc-events.md:141]: Why not add these to the global excludes? They're commonly used.

File: design/baremetal-operator/how-ironic-works.md [hunk 3]
```diff
@@ -7,8 +7,10 @@
 
 # how-ironic-works
 
+<!-- cSpell:ignore irmc,xclarity,uncompress -->
```
Comment (dtantsur) [inline design/baremetal-operator/how-ironic-works.md:10]: Should we replace uncompress with decompress?

File: design/baremetal-operator/image-builder-integration.md [hunk 4]
```diff
@@ -7,6 +7,8 @@
 
 # Custom agent image controller
 
+<!-- cSpell:ignore statelessly -->
```
Comment (dtantsur) [inline design/baremetal-operator/image-builder-integration.md:10]: nit: maybe we should use "in a stateless way" or something like that

File: design/baremetal-operator/reboot-interface.md [hunk 5]
```diff
@@ -7,6 +7,8 @@
 
 # reboot-interface
 
+<!-- cSpell:ignore rebalancing,accomodate,unprovisioned,suffixless -->
```
Comment (dtantsur) [inline design/baremetal-operator/reboot-interface.md:10]: accomodate looks wrong to me

File: design/baremetal-operator/user-defined-root-device-hints.md [hunk 6]
```diff
@@ -13,7 +13,7 @@ implemented
 
 ## Summary
 
-This document explains how rootdevicehints could be given as user
+This document explains how `rootdevicehints` could be given as user
```
Comment (dtantsur) [inline design/baremetal-operator/user-defined-root-device-hints.md:16]: nit: `rootDeviceHints` to match the actual capitalization in the JSON body

File: design/cluster-api-provider-metal3/allow_disabling_node_disk_cleaning.md [hunk 7]
```diff
@@ -7,6 +7,8 @@ http://creativecommons.org/licenses/by/3.0/legalcode
 
 # allow disabling automated cleaning
 
+<!-- cSpell:ignore dtantsur -->
```
Comment (dtantsur) [inline design/cluster-api-provider-metal3/allow_disabling_node_disk_cleaning.md:10]: Don't ignore me :(  /jk

**Comment by dtantsur** [comment]
Wow, it's a lot of work! I left some comments but I'll be happy with a follow-up.

/approve
/lgtm

File: design/baremetal-operator/bmc-events.md (hunk 1 above)
Comment (tuminoid) [reply design/baremetal-operator/bmc-events.md:82]: I can change that back if its an actual term and not just common shorthand.

File: design/baremetal-operator/bmc-events.md (hunk 2 above)
Comment (tuminoid) [reply design/baremetal-operator/bmc-events.md:141]: I probably added them, but I missed removing temporary ignores here.

File: design/baremetal-operator/image-builder-integration.md (hunk 4 above)
Comment (tuminoid) [reply design/baremetal-operator/image-builder-integration.md:10]: Sure, I did not want to do editorials in such a big PR, I can fix that in follow-up.

File: design/baremetal-operator/reboot-interface.md (hunk 5 above)
Comment (tuminoid) [reply design/baremetal-operator/reboot-interface.md:10]: You mean it should not be ignored, or there is some other spelling for it?

File: design/baremetal-operator/user-defined-root-device-hints.md (hunk 6 above)
Comment (tuminoid) [reply design/baremetal-operator/user-defined-root-device-hints.md:16]: I'll fix that.

File: design/baremetal-operator/how-ironic-works.md (hunk 3 above)
Comment (tuminoid) [reply design/baremetal-operator/how-ironic-works.md:10]: Done

File: design/baremetal-operator/reboot-interface.md (hunk 5 above)
Comment (tuminoid) [reply design/baremetal-operator/reboot-interface.md:10]: Ah, it can have double m.  Fixed.

**Comment by tuminoid** [comment]
> Wow, it's a lot of work! I left some comments but I'll be happy with a follow-up.

Follow-up here: https://github.com/metal3-io/metal3-docs/pull/482

**Pull request merged**